source/grid/grid_generator.cc

```diff
@@ -3091,7 +3089,32 @@ namespace GridGenerator
       {
         Assert(false, ExcIndexRange(n, 0, 5));
       }
+    if (colorize)
+      {
+        typename Triangulation<3>::cell_iterator cell = tria.begin();
+        for (; cell!=tria.end(); ++cell)
+          for (unsigned int i=0; i<GeometryInfo<3>::faces_per_cell; ++i)
+            if (cell->at_boundary(i))
+              cell->face(i)->set_all_boundary_ids(2);
+
+        for (cell=tria.begin(); cell!=tria.end(); ++cell)
+          for (unsigned int i=0; i<GeometryInfo<3>::faces_per_cell; ++i)
+            if (cell->at_boundary(i))
+              {
+                const typename Triangulation<3>::face_iterator face
+                  = cell->face(i);
 
+                const Point<3> face_center (face->center());
+                if (std::abs((face_center(0)-center(0)) > 1.e-3))
+                  {
+                    if (std::abs((face_center-center).norm()-inner_radius) <
+                        std::abs((face_center-center).norm()-outer_radius))
```

Comment:
I find these nested ifs hard to follow. Can you add a comment how they are supposed to work?